packages/api/tests/app/test-app.js

```diff
@@ -1,22 +1,28 @@
+/* eslint-disable global-require */
 const test = require('ava');
 
 const { MissingRequiredEnvVar } = require('@cumulus/errors');
+const { secretsManager } = require('@cumulus/aws-client/services');
+const { randomString } = require('@cumulus/common/test-utils');
 
-const { handler } = require('../../app');
+let handler;
 
 test.beforeEach(() => {
   process.env.dynamoTableNamesParameterName = 'fake-param-name';
+  delete require.cache[require.resolve('../../app')];
 });
 
 test.serial('handler throws error if environment variable for Dynamo tables parameter name is missing', async (t) => {
+  ({ handler } = require('../../app'));
```

Comment:
Is this for forcing the environment variables to be re-evaluated every time?

Reply:
Yes, in concert with L12

Reply:
Can't we just unit test `initEnvVarsFunction` separately (and perhaps should)? And then we just need one test to prove that any variables from a secret that exists before `await handler` is called will be populated into `process.env`? I'm not sure if your last test here would need to change much

I'm a little wary of messing with the `require` cache especially if/when we want to go to ES6 modules

Reply:
Hmm, that last point is compelling.      I'll split it into a separate file.

Reply:
f8f97e304